packages/vsce/src/utils/commandUtils.ts

```diff
@@ -13,6 +13,148 @@ import { TreeView } from "vscode";
 import { IResourceMeta } from "../doc";
 import { CICSRegionsContainer, CICSResourceContainerNode } from "../trees";
 import { IResource } from "@zowe/cics-for-zowe-explorer-api";
+import { IProfileLoaded } from "@zowe/imperative";
+import { Gui, ZoweExplorerApiType, ZoweVsCodeExtension } from "@zowe/zowe-explorer-api";
+import * as vscode from "vscode";
+import { CICSLogger } from "./CICSLogger";
+import { ProfileManagement } from "./profileManagement";
+
+/**
+ * Checks if a profile supports a specific type of connection
+ *
+ * @param profile - The profile to check
+ * @param connectionType - The type of connection to check for (ZoweExplorerApiType.Uss or ZoweExplorerApiType.Jes)
+ * @returns True if the profile supports the specified connection type, false otherwise
+ */
+export function doesProfileSupportConnectionType(profile: IProfileLoaded, connectionType: ZoweExplorerApiType): boolean {
+  const explorerApi = ZoweVsCodeExtension.getZoweExplorerApi();
+
+  try {
+    switch (connectionType) {
+      case ZoweExplorerApiType.Uss:
+        explorerApi.getUssApi(profile);
+        break;
+      case ZoweExplorerApiType.Jes:
+        explorerApi.getJesApi(profile);
+        break;
+      default:
+        return false;
+    }
+    return true;
+  } catch {
+    CICSLogger.error(`Profile ${profile.name} does not support connection type ${connectionType}`);
+    return false;
+  }
+}
+
+/**
+ * Fetches a base profile without throwing an error if none exists
+ *
+ * @param profile - The profile to fetch the base profile for
+ * @returns The base profile or undefined if none exists
+ */
+export async function fetchBaseProfileWithoutError(profile: IProfileLoaded): Promise<IProfileLoaded | undefined> {
+  let baseProfile = undefined;
+  try {
+    baseProfile = await ProfileManagement.getProfilesCache().fetchBaseProfile(profile.name);
+  } catch (ex) {
+    CICSLogger.error(`No base profile found for ${profile.name}`);
```

Comment:
I don't think we should output this as an error, because we are using this function to check whether a base profile exists or not. If it doesn't exist, that's not an error, just "no base profile".

```suggestion
    // this isn't an error we're interested in - we were checking if a base profile existed
    CICSLogger.debug(`No base profile found for ${profile.name}`);
```